### PR TITLE
Feat/met 3412 develop progress tracking component

### PR DESCRIPTION
**Pull request by andyjmaclean**

### CODE REVIEW

File: projects/sandbox/src/app/wizard/wizard.component.conf.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
+/** Defines steps for upload
```
[the remark this answers is not in the document]
Comment (andyjmaclean) [reply projects/sandbox/src/app/wizard/wizard.component.conf.ts]: This convention is already in place (see filter-options-conf) but the point is irrelevant now as I've removed this file following some refactoring.  As a result all copy is in the template and no @Inputs are required.  A smaller configuration object has been added directly to the wizard component to group the fields for selective validation.

File: projects/sandbox/src/app/_mocked/mocked-progres-info.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { DatasetInfo, ProgressByStep } from '../_models';
```
[the remark this answers is not in the document]
Comment (andyjmaclean) [reply projects/sandbox/src/app/_mocked/mocked-progres-info.ts:1]: Oops.  Fixed.

File: projects/sandbox/src/app/wizard/wizard.component.ts [hunk 3]
```diff
@@ -113,51 +161,86 @@ export class WizardComponent extends DataPollingComponent {
...
 
+  /**
+   * progressComplete
+   * Template utility to determine if the progress is complete
+   *
+   **/
+  progressComplete(): boolean {
+    return this.progressData && this.progressData.status === DatasetInfoStatus.COMPLETED
+      ? true
+      : false;
```
[the remark this answers is not in the document]
Comment (andyjmaclean) [reply projects/sandbox/src/app/wizard/wizard.component.ts]: AndyError101: too many ternaries!

File: projects/sandbox/src/app/wizard/wizard.component.ts [hunk 4]
```diff
@@ -72,31 +75,76 @@ export class WizardComponent extends DataPollingComponent {
...
+  getFormGroup(stepConf: WizardStep): FormGroup {
+    return stepConf.stepType === WizardStepType.PROGRESS_TRACK
+      ? this.formProgress
+      : this.formUpload;
+  }
+
+  /**
+   * getStepIsSubmittable
+   *
+   * @param { number } step - the index of the WizardStep to evaluate
```
[the remark this answers is not in the document]
Comment (andyjmaclean) [reply projects/sandbox/src/app/wizard/wizard.component.ts]: No it isn't. Fixed.

File: projects/sandbox/src/app/wizard/wizard.component.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jochen-vermeulen) [reply projects/sandbox/src/app/wizard/wizard.component.ts]: :)

File: projects/sandbox/src/app/wizard/wizard.component.ts [hunk 5]
```diff
@@ -1,19 +1,20 @@
...
   SubmissionResponseData,
-  WizardField,
   WizardStep,
   WizardStepType
 } from '../_models';
 import { SandboxService } from '../_services';
-import { ProtocolType } from '@shared';
+import { ProtocolType } from 'shared';
 
 export interface ValidatorArrayHash {
```
Comment (jochen-vermeulen) [inline projects/sandbox/src/app/wizard/wizard.component.ts:19]: According to my IDE this is now unused.

File: projects/sandbox/src/app/wizard/wizard.component.ts [hunk 6]
```diff
@@ -57,37 +82,46 @@ export class WizardComponent extends DataPollingComponent {
...
-              }
-              map[wf.name] = entry;
-              return map;
-            }, {});
-          return Object.assign(map, test);
-        }, {})
-    );
+
+    this.formUpload = this.fb.group({
+      name: ['', [Validators.required, this.validatorWhitespace]],
```
Comment (jochen-vermeulen) [inline projects/sandbox/src/app/wizard/wizard.component.ts:89]: I think the name requires some more validation. Will this be part of a separate subsequent ticket?

Comment (andyjmaclean) [reply projects/sandbox/src/app/wizard/wizard.component.ts:89]: Yes - full validation would have been out of scope for this ticket!

File: projects/sandbox/src/app/wizard/wizard.component.ts (hunk 5 above)
Comment (andyjmaclean) [reply projects/sandbox/src/app/wizard/wizard.component.ts:19]: It is used.  You probably need to re-build the shared library.

**Pull request merged**